core/frontend-devtools/src/tools/ViewDefinitionDecorator.ts

```diff
@@ -0,0 +1,134 @@
+ */
+
+import { ColorDef, Frustum, LinePixels } from "@itwin/core-common";
+import { DecorateContext, GraphicType, IModelApp, IModelConnection, OffScreenViewport, Tool, Viewport, ViewRect, ViewState } from "@itwin/core-frontend";
+import { parseToggle } from "./parseToggle";
+import { Id64, Id64String } from "@itwin/core-bentley";
+import { Transform, Vector2d } from "@itwin/core-geometry";
+
+class ViewDefinitionDecoration {
+  private static _instance?: ViewDefinitionDecoration;
+  private _removeMe?: () => void;
+
+  private constructor() {
+    this._removeMe = IModelApp.viewManager.addDecorator(this);
+  }
+
+  private stop() {
+    if (this._removeMe) {
+      this._removeMe();
+      this._removeMe = undefined;
+    }
+  }
+
+  private _viewId?: Id64String;
+  private _loading: boolean = false;
+  private _preloadedFrustum: Frustum[] = [];
+
+  private async getFrustumForViewId(iModel: IModelConnection, viewId: Id64String, transform: Transform = Transform.createIdentity()): Promise<ViewState> {
+    return iModel.views.load(viewId).then((result) => {
+      // Use the view extents/delta
+      let vector: Vector2d;
+      if (result.is2d()) vector = Vector2d.create(result.delta.x, result.delta.y);
+      else vector = Vector2d.create(result.getExtents().x, result.getExtents().y);
+      // using new ViewRect(0,0,1,1) to copy SectionAttachment
+      const vp = OffScreenViewport.create({ view: result, viewRect: new ViewRect(0,0,vector.x,vector.y) });
+      const frustum = vp.getFrustum();
+      this._preloadedFrustum.push(frustum.transformBy(transform));
+
+      return result;
+    })
+  }
+
+  public preload(viewport: Viewport) {
+    this._loading = true;
+    const iModel = viewport.iModel;
+    const view = viewport.view;
+    const viewId = view.id;
+
+    const loadPromises: Promise<ViewState>[] = [];
+    loadPromises.push(this.getFrustumForViewId(iModel, viewId));
+
+    if (view.isDrawingView() && view.sectionDrawingInfo.spatialView)
+      loadPromises.push(this.getFrustumForViewId(iModel, view.sectionDrawingInfo.spatialView, view.sectionDrawingInfo.drawingToSpatialTransform.inverse()));
+
+    void Promise.all(loadPromises).finally(() => {
+      IModelApp.viewManager.invalidateCachedDecorationsAllViews(this);
+      this._viewId = viewId;
+      this._loading = false
+    });
+  }
+
+  /** This will allow the render system to cache and reuse the decorations created by this decorator's decorate() method. */
+  public readonly useCachedDecorations = true;
+
+  public decorate(context: DecorateContext): void {
+    if (context.viewport.view.id !== this._viewId && !this._loading) {
+      this._preloadedFrustum.length = 0;
+      // just clear the decoration if the view is not from a view definition
+      if (Id64.isInvalid(context.viewport.view.id)) return;
+      try {
+        this.preload(context.viewport);
+      } catch (err: unknown) {
+        // eslint-disable-next-line no-console
+        console.log(err);
+      }
```

Comment:
```suggestion
      this.preload(context.viewport);
```

Do you frequently observe exceptions here? If so what kind?